curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/store-multiple-values-in-one-variable-using-javascript-arrays.md

```diff
@@ -19,23 +19,23 @@ var sandwich = ["peanut butter", "jelly", "bread"]
 
 # --instructions--
 
-Modify the new array `myArray` so that it contains both a `string` and a `number` (in that order).
+Modify the new array `myArray` so that it contains both a string and a number (in that order).
 
 # --hints--
 
-`myArray` should be an `array`.
+`myArray` should be an array.
 
 ```js
 assert(typeof myArray == 'object');
 ```
 
-The first item in `myArray` should be a `string`.
+The first item in `myArray` should be a string.
 
 ```js
 assert(typeof myArray[0] !== 'undefined' && typeof myArray[0] == 'string');
```

Comment:
I would just like clarification on these changes.

My understanding is, as long as something refers to an English keyword/variable/value, it is best to not translate it. For this, the tests are looking for a keyvalue `string`. How would something like this be translated?:
> The first item should be a BigInt.

I assume there is no direct translation for BigInt, in most languages?

Reply:
I think both could be used depending how text is written, for example something like this:
>`typeof myArray[0]` should return `string`

but, as in the case above:
>The first item in `myArray` should be a string.

Reply:
That was my understanding, that if we're referring to a specific bit of code, it should be in code tags. 

It can be a grey area, though, when you're being somewhat informal.

Reply:
The thing is, it's the nuances of the language that we would want/need to work around. If something feels more natural with a keyword translated in the world language and the proofreader makes an expert decision, it should be OK.

I think, in general, we have been removing backticks quite a bit, where there is an overlap between keywords. A lot would depend on the context, of course. 

The bottom-line is the sentence should be as natural sounding as possible to get the point across.

But, I will let @ieahleen confirm if this correct.

Reply:
Here is a list of the words that are and are not being translated into Italian @ShaunSHamilton 
https://docs.google.com/spreadsheets/d/14oYUuREk3Qv8ewI-MVqd-DUT2QmmmDnq4aie8oyCt0Q/edit#gid=0

Reply:
@Sembauke Thank you, for linking to that 💯

Reply:
If the preference is on keeping the data types in backticks, and so untranslated, it should be uniformed in all the curriculum
but the data types are presented the first time in backticks,
here all of them: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-javascript-variables
here the Number data type: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/add-two-numbers-with-javascript
and then later as normal words
first time that a test says that something should be a number is not in backticks:
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/create-decimal-numbers-with-javascript

The second situation is far more common

But if the changes are not wanted we can use the same thing that have been done for spanish and chinese, remove the untraslatable word from the sentence flow

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/store-multiple-values-in-one-variable-using-javascript-arrays.md
https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/espanol/02-javascript-algorithms-and-data-structures/basic-javascript/store-multiple-values-in-one-variable-using-javascript-arrays.md

Reply:
> If the preference is on keeping the data types in backticks, and so untranslated, it should be uniformed in all the curriculum

As I do little-to-no translating, I have no preference, but I agree with trying to keep uniformity. This is why I would like a translator's preference, because we are in the perfect position to get uniformity with at least the new curriculum.

I suppose, as long as the translators have a dictionary like Sem linked, then, in most cases I would follow @gikf 's  example

Reply:
What happens at the moment if you try to save a string with the code tags removed?  If memory serves it should warn you, but you should be able to save anyway.  I didn't want to submit any bad translations just to test this, though.

Isn't that behaviour we want to keep?  Reason being, if the translator and proofreaders agree that it would be a worse translation with the code tags, then they can ignore the warning and approve the translation.  Also we keep the guidance that, in general, we think code tags should be there.

Having said all that, we should definitely remove code tags if they're inappropriate (just used for decoration, say).

Reply:
It can be saved anyway
![image](https://user-images.githubusercontent.com/26656284/121304278-6f53aa80-c8fc-11eb-9941-f69cf22d20e2.png)

Reply:
Cool, that's good.  How about this for a plan

- Update the English if the code tags don't make sense
- Update the [translation best practices](https://contribute.freecodecamp.org/#/how-to-translate-files?id=translation-best-practices) to say something like "Typically text inside `<code>` tags should not be translated, but if doing so would result in a bad translation, the warnings can be ignored"

@ieahleen, you're actually translating, unlike me, does that seem reasonable?

Reply:
yes, totally, just need to explain to @RafaelDavisH what's going on, as he is often doing the onboarding of new translators

Reply:
My understanding also is, code tags are being used not only prevent translating programming language reversed keywords and words or phrases that are part of the challenge, and are tested. So, if we have code tags wrapping words or phrases that don't affect in any way any test, then I don't think we have them. 

When we (ES proofreaders) have encounter code tags wrapping words that by having them in English the sentence don't make sense, we have added the translation next to the wrapped word. I am sure this method is not always doable.


> "Typically text inside `<code>` tags should not be translated, but if doing so would result in a bad translation, the warnings can be ignored"

Having translators to decide if whether they should translate the wrapped words could arise in many instances problems that the proofreaders will have to resolved. Not all translators are programmers.

We have either enforce NOT translated wrapped keywords or NOT enforce at all. Having devs also helping with the translators as @ieahleen will help clean up those unnecessary wrapped words case by case.

Reply:
I'm not a fan of taking the "save anyway" approach, or the "don't translate usually, but sometimes it's okay".

IMHO, opening things up to a "do what you think is best" approach will lead to inconsistencies in our translation style and UX. We should instead implement a process to review the items in code tags and allow translators to flag items that should be "un-coded".

Reply:
However, I think this merged PR is not the place for this discussion. I'll migrate the conversation to a new issue.